Tests/SwiftStdlibTests/SequenceExtensionsTests.swift

```diff
@@ -151,5 +151,10 @@ final class SequenceExtensionsTests: XCTestCase {
         let array2 = ["James", "Wade", "Bryant", ""]
         XCTAssertEqual(array2.sorted(by: \String.first, with: optionalCompare), ["Bryant", "James", "Wade", ""])
     }
+    
+    func testKeyPathGrouped() {
+        let array = ["James", "Wade", "Bryant", "John"]
```

Comment:
Could you add an empty sequence case?

Reply:
@LucianoPAlmeida added a few extra cases actually, please take a look.

I am a little unhappy with how optional keypaths get handled, but can't think of a clean solution of the top of my head.  I will probably mess with it later, unless you got something obvious / easy i am missing or forgetting.

Reply:
I don't see a problem with optionals here... to me seems the expected behavior :))